test/Test/Property.hs

```diff
@@ -199,6 +207,32 @@ alternativeLeftIdentity = hedgehog $ do
     x <- forAll $ genValidation genSmallText
     (empty <|> x) === x
 
+----------------------------------------------------------------------------
+-- Selective instance properties
+----------------------------------------------------------------------------
+
+selectiveIdentity :: Property
+selectiveIdentity = do
+    x <- forAll $ genValidation $ genEither genSmallText genSmallText
+    (x <*? pure id) === (either id id <$> x)
+
+selectiveDistributivity :: Property
+selectiveDistributivity = do
+    x <- forAll $ genEither genInt genInt
+    y <- forAllWith (const "y") $ genValidation genFunction
+    z <- forAllWith (const "z") $ genValidation genFunction
+    (pure x <*? (y *> z)) === ((pure x <*? y) *> (pure x <*? z))
+
+selectiveAssociativity :: Property
+selectiveAssociativity = do
+    x <- forAllWith (const "x") $ genValidation $ genEither genInt genInt
```

Comment:
It seems that we don't need `forAllWith` here because this `Validation` doesn't contain a function
```suggestion
    x <- genValidation $ genEither genInt genInt
```